### PR TITLE
NEXT-8288

### PR DESCRIPTION
**Pull request by kiplingi**
# 1. Why is this change necessary?
GDPR Compliance

### 2. What does this change do, exactly?
    * Removed `required` status for `firstName` and `lastName`on `newsletter-form.html.twig`
    * Added fallback for missing `getFirstName` and `getLastName` on `NewsletterRegisterEvent.php`

### 4. Please link to the relevant issues (if any).
https://github.com/shopware/platform/issues/819
[shopware.zip](https://github.com/shopware/platform/files/4629096/shopware.zip)

### CODE REVIEW

**Comment by mitelg** [comment]
hey @kiplingi 
please squash your commits and adjust the commit message just like shown in the document provided in the slack channel.

**Comment by mitelg** [comment]
also your branch needs a rebase :wink:

File: src/Core/Content/Newsletter/Event/NewsletterRegisterEvent.php [hunk 1]
```diff
@@ -83,9 +83,17 @@ public function getMailStruct(): MailRecipientStruct
...
         }
 
+        if ($this->newsletterRecipient->getFirstName() && $this->newsletterRecipient->getLastName()) {
+            return new MailRecipientStruct(
+                [
+                    $this->newsletterRecipient->getEmail() => $this->newsletterRecipient->getFirstName() . ' ' . $this->newsletterRecipient->getLastName(),
+                ]
+            );
+        }
+
```
Comment (shyim) [inline src/Core/Content/Newsletter/Event/NewsletterRegisterEvent.php]:
```suggestion
$recipientName = $this->newsletterRecipient->getEmail();

        if ($this->newsletterRecipient->getFirstName() && $this->newsletterRecipient->getLastName()) {
            $recipientName =  $this->newsletterRecipient->getFirstName() . ' ' . $this->newsletterRecipient->getLastName();
            );
        }

```

Comment (shyim) [reply src/Core/Content/Newsletter/Event/NewsletterRegisterEvent.php]: What do you think about this? and using the variable in the last line

Comment (tinect) [reply src/Core/Content/Newsletter/Event/NewsletterRegisterEvent.php]: @shyim you want to send mail to `mail@example.com <mail@example.com>` sounds creapy :-)

File: CHANGELOG-6.2.md [hunk 2]
```diff
@@ -574,6 +574,20 @@ To get the diff between two versions, go to https://github.com/shopware/platform
     * Deprecated `sort` parameter for product listing, search and suggest gateway, use `order` instead
     * Added block `document_line_item_table_iterator` to `@Framework\documents\base.html.twig` to override the lineItem iterator
     * Added `StoreApiClient` which allows to send requests to `store-api` and `sales-channel-api` routes.
+    * Removed `required` status for `firstName` and `lastName`on `newsletter-form.html.twig`
+    * Added fallback for missing `getFirstName` and `getLastName` on `NewsletterRegisterEvent.php`
+    * Added new Twig blocks in `src/Storefront/Resources/views/storefront/page/account/order/index.html.twig`
```
Comment (mitelg) [inline CHANGELOG-6.2.md:579]: please do again a proper rebase :blush: seems that you messed the changelog up :see_no_evil:

**Review by mitelg — CHANGES_REQUESTED** [review]

File: CHANGELOG-6.2.md (hunk 2 above)
Comment (kiplingi) [reply CHANGELOG-6.2.md:579]: On it :see_no_evil:

File: src/Core/Content/Newsletter/Event/NewsletterRegisterEvent.php (hunk 1 above)
Comment (kiplingi) [reply src/Core/Content/Newsletter/Event/NewsletterRegisterEvent.php]: @shyim thanks, integrated :)
@tinect But that's the plan. The thing is, the mailing service wants to set the recipient properly, but when the only information is the email we thought it better to be that then an empty string.

**Comment by kiplingi** [comment]
I made a new pull request for this: https://github.com/shopware/platform/pull/936

**Pull request closed without merging**